### PR TITLE
71a savegame parser

### LINKED ISSUE #104: Parse saved json to State class
Parse saved jsonto State class

### PR DESCRIPTION
**Pull request by MathiasMeuleman**
Will close #104 

The first parser is ready and is integrated in the start of the game.

<feature

### CODE REVIEW

**Comment by MKerkhof** [comment]
Still has three easy to fix PMD warnings like useless parenthesis, and a lot of checkstyle errors. Besides that, didn't spot anything that's not oke.

<review

File: src/main/java/nl/tudelft/ti2206/group9/gui/SettingsScene.java [hunk 1]
```diff
@@ -37,7 +35,7 @@
      * @return whether sound is enabled.
      */
     public static boolean isSoundEnabled() {
-    	return sound;
+    	return State.isSoundEnabled();
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/gui/SettingsScene.java]: If you make it return State.isSoundEnabled(), why does this method still exist?

<interface,review

File: src/main/java/nl/tudelft/ti2206/group9/level/State.java [hunk 2]
```diff
@@ -127,4 +133,19 @@ public static int getHighscore() {
...
+		State.playerName = newPlayerName;
+	}
+
+	public static boolean isSoundEnabled() {
+		return soundEnabled;
+	}
+
+	public static void setSoundEnabled(boolean newSoundEnabled) {
+		State.soundEnabled = newSoundEnabled;
+	}
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/level/State.java]: These setters and getters are missing JavaDoc.

<documentation,review

File: src/main/java/nl/tudelft/ti2206/group9/util/SaveGameParser.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import org.json.simple.parser.ParseException;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+
+/**
+ * @author Mathias
+ */
+public class SaveGameParser {
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/util/SaveGameParser.java]: *Auto-generated comment*

    Utility classes should not have a public or default constructor.

File: src/main/java/nl/tudelft/ti2206/group9/util/SaveGameParser.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+	/**
+	 * Load all data from the given file and save it in State.
+	 * @param filePath the path to the file to be read.
+	 */
+	public static final void loadGame(final String filePath) {
+		try {
+			JSONObject mainObject = parserInit(filePath);
+			playername = (String) mainObject.get("playername");
+			coins = Integer.valueOf((String) mainObject.get("coins"));
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/util/SaveGameParser.java]: *Auto-generated comment*

    Boxing/unboxing to parse a primitive nl.tudelft.ti2206.group9.util.SaveGameParser.loadGame(String)

File: src/main/java/nl/tudelft/ti2206/group9/util/SaveGameParser.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+	/**
+	 * Initialize the parsing process.
+	 * @param filePath path to the file to be parsed.
+	 * @return JSONObject the main JSONObjct from the read file
+	 * @throws IOException IOException
+	 * @throws ParseException ParseException
+	 */
+	private static JSONObject parserInit(final String filePath) throws IOException, ParseException {
+		FileReader reader = new FileReader(filePath);
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/util/SaveGameParser.java]: Findbugs doesn't like the way you read the file. 

<review

File: src/test/java/nl/tudelft/ti2206/group9/level/StateTest.java [hunk 6]
```diff
@@ -57,7 +57,7 @@ public void testModulo_Score() {
 	}
 	
 	@Test
-	public void testGetDistance() {
+	public void tesSGetDistance() {
```
Comment (mpsijm) [inline src/test/java/nl/tudelft/ti2206/group9/level/StateTest.java]: Yay typos :D 
<naming

File: src/main/java/nl/tudelft/ti2206/group9/util/SaveGameParser.java (hunk 5 above)
Comment (MathiasMeuleman) [inline src/main/java/nl/tudelft/ti2206/group9/util/SaveGameParser.java]: *Auto-generated comment*

    nl.tudelft.ti2206.group9.util.SaveGameParser.parserInit(String) may fail to clean up java.io.Reader

File: src/main/java/nl/tudelft/ti2206/group9/util/SaveGameParser.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+	/** Boolean to indicate whether the sound is enabled. */
+	private static boolean soundEnabled;
+
+	/**
+	 * Read a json savefile and store all data in the State class.
+	 * @param filePath path to the file to be parsed.
+	 */
+	public static void loadGame(final String filePath) {
+		try {
+			FileReader reader = new FileReader(filePath);
```
Comment (MathiasMeuleman) [inline src/main/java/nl/tudelft/ti2206/group9/util/SaveGameParser.java]: *Auto-generated comment*

    Found reliance on default encoding in nl.tudelft.ti2206.group9.util.SaveGameParser.loadGame(String): new java.io.FileReader(String)

**Comment by mdingjan** [comment]
Could you have one more look at SaveGamerParserTest? There are a few unused imports left and a line that's longer than 80 characters.

<review,organisation

**Comment by mdingjan** [comment]
Now there are no checkstyle warnings left. :) Will merge!

<review

**Pull request merged**